### ISSUE TITLE
Regex matching for text

### ISSUE THREAD

**Issue by eeue56** [eeue56/elm-html-test#58]
I found myself in the situation where I'd like to be able to match against part of a string within a text field. 

For example:

https://github.com/terezka/line-charts/pull/38/files#diff-c75cfaadb0c1fe89dc19f42360af0317R91

In this case, the "10" part of the UI doesn't actually matter: what does matter is the "Age"

**Comment by rtfeldman** [comment]
Makes sense to me! I'm familiar with this use case from Capybara tests. 

What about this for an API?

```elm
textMatches : (String -> Bool) -> Selector

textContains : String -> Selector
```

The former is maximally flexible - you could use regex, or a helper function, or `Parser`, or anything else.

The latter would be the most common application of that function. (So you wouldn't have to say `textMatches (Sting.contains "Age")` all the time.)

**Comment by rtfeldman** [comment]
I just realized that my proposed `textContaining` is actually how `text` currently works.

This means it's not currently possible to search for (for example) an empty text node, e.g. `text ""` doesn't do anything useful.

This seems like a useful distinction to make!
